packit/git_utils.py

```diff
@@ -0,0 +1,109 @@
+# MIT License
+#
+# Copyright (c) 2018-2020 Red Hat, Inc.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Helpers and wrappers on top of git.
+"""
+import logging
+from typing import Optional
+
+import git
+import yaml
+
+from packit.exceptions import PackitException
+
+logger = logging.getLogger(__name__)
+
+
+def get_message_from_metadata(metadata: dict, header: Optional[str] = None) -> str:
+    if not isinstance(metadata, dict):
+        raise PackitException(
+            f"We can save only dictionaries to metadata. Not {metadata}"
+        )
+
+    content = yaml.dump(metadata, indent=4) if metadata else ""
+    if not header:
+        return content
+
+    return f"{header}\n\n{content}"
+
+
+def get_metadata_from_message(commit: git.Commit) -> Optional[dict]:
+    """
+    Tries to load yaml format from the git message.
+
+    We are skipping first line until
+    the rest of the content is yaml-loaded to dictionary (yaml object type).
```

Comment:
The problem here are the commit message suffixes: `Approved-by`, `Signed-off` and all of these...

Reply:
It's not a problem, we will load that as a following metadata:

```
{
  "Approved-by" : "someone"
  " Signed-off": "someone else"
}
```

It will not harm anything, won't it?

Reply:
nice! yaml ftw